git/pre-commit

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -eu
+
+# This file will exist in $project_root/.git/hooks/, so we need to move back up
+# to $project_root. This apparent mess needs to happen because macOS doesn't
+# have `readlink -f`.
+declare src="$0"
+declare dir=''
+
+while [ -h "$src" ]; do
+  dir="$(cd -P "$( dirname "$src")" && pwd)"
+  src="$(readlink "$src")"
+  [[ "$src" != /* ]] && src="$dir/$src"
+done
+
+cd -P "$(dirname "$src")/../../"
+
+##### This is the actual pre-commit hook. #####
```

Comment:
We should either explain the new requirements (pip install requirements/securedrop-development.txt) or we could just not run make lint if the requirements are not satisfied.

I'm in favor of making this pre-commit hook optional rather than mandatory, therefore leaning toward not doing anything if the requirements are not met.

Reply:
If you don't want to run it, just don't copy it into `.git`. Or use `--no-verify`. :)

Reply:
Oh, I wrongly assumed it would be installed by default! Sorry for the noise. I should have read the guidelines :-)